include/fiction/algorithms/simulation/sidb/sidb_simulation_engine.hpp

```diff
@@ -169,11 +177,19 @@ template <typename EngineType>
 [[nodiscard]] inline std::optional<sidb_simulation_engine>
 get_sidb_simulation_engine(const std::string_view& name) noexcept
 {
+#if (FICTION_ALGLIB_ENABLED)
     static const phmap::flat_hash_map<std::string, sidb_simulation_engine> engine_lookup{
         {"EXGS", sidb_simulation_engine::EXGS},
         {"QUICKEXACT", sidb_simulation_engine::QUICKEXACT},
         {"CLUSTERCOMPLETE", sidb_simulation_engine::CLUSTERCOMPLETE},
         {"QUICKSIM", sidb_simulation_engine::QUICKSIM}};
+#else   // FICTION_ALGLIB_ENABLED
+
+    static const phmap::flat_hash_map<std::string, sidb_simulation_engine> engine_lookup{
+        {"EXGS", sidb_simulation_engine::EXGS},
+        {"QUICKEXACT", sidb_simulation_engine::QUICKEXACT},
+        {"QUICKSIM", sidb_simulation_engine::QUICKSIM}};
+#endif  // FICTION_ALGLIB_ENABLED
```

Comment:
Not entirely sure, but it looks like this can be simplified.
```suggestion
    static const phmap::flat_hash_map<std::string, sidb_simulation_engine> engine_lookup{
        {"EXGS", sidb_simulation_engine::EXGS},
        {"QUICKEXACT", sidb_simulation_engine::QUICKEXACT},
#if (FICTION_ALGLIB_ENABLED)
        {"CLUSTERCOMPLETE", sidb_simulation_engine::CLUSTERCOMPLETE},
#endif   // FICTION_ALGLIB_ENABLED
        {"QUICKSIM", sidb_simulation_engine::QUICKSIM}};
```

Reply:
It should also get rid of the `clang-tidy` warning.